### PR TITLE
Made changes according to the commit RD-4036 of sdk-java

### PR DESCRIPTION
**Pull request by priyanshu-ap**

### CODE REVIEW

File: tpp/src/main/Member.js [hunk 1]
```diff
@@ -483,4 +486,42 @@ export default class Member extends CoreMember {
...
+
+    /**
+     * Get the raw consent from the bank associated with a token.
+     *
+     * @param tokenId {string} Token Id
+     * @returns {GetRawConsentResponse} raw consent
+     */
+    getRawConsent(tokenId: string): Promise<GetRawConsentResponse> {
+        return Util.callAsync(this.getBankAuthUrl, async () => {
+            await this._client.onBankAuthCallback(tokenId);
```
Comment (grekinsky) [inline tpp/src/main/Member.js]: This should be `getRawConsent`

File: tpp/src/main/Member.js [hunk 2]
```diff
@@ -483,4 +486,42 @@ export default class Member extends CoreMember {
...
+    }
+
+    /**
+     * Get the raw consent from the bank associated with a token.
+     *
+     * @param tokenId {string} Token Id
+     * @returns {GetRawConsentResponse} raw consent
+     */
+    getRawConsent(tokenId: string): Promise<GetRawConsentResponse> {
+        return Util.callAsync(this.getBankAuthUrl, async () => {
```
Comment (grekinsky) [inline tpp/src/main/Member.js]: should be `getRawConsent`

File: tpp/src/main/Member.js [hunk 3]
```diff
@@ -483,4 +486,42 @@ export default class Member extends CoreMember {
...
+
+    /**
+     * Forward the callback from the bank (after user authentication) to Token.
+     *
+     * @param bankId {string} Bank Id
+     * @param query {string} HTTP query string
+     * @returns {OnBankAuthCallbackResponse} token request ID
+     */
+    onBankAuthCallback(bankId: string, query: string): Promise<OnBankAuthCallbackResponse> {
+        return Util.callAsync(this.getBankAuthUrl, async () => {
```
Comment (grekinsky) [inline tpp/src/main/Member.js]: this should be `onBankAuthCallback`

File: tpp/src/main/Member.js [hunk 4]
```diff
@@ -483,4 +486,42 @@ export default class Member extends CoreMember {
...
     }
+
+    /**
+     * Get url to bank authorization page for a token request.
+     *
+     * @param bankId {string} Bank Id
+     * @param tokenRequestId {string} Token Request Id
+     * @returns {GetBankAuthUrlResponse} url
+     */
+    getBankAuthUrl(bankId: string, tokenRequestId: string): Promise<GetBankAuthUrlResponse> {
```
Comment (gianluca-token) [inline tpp/src/main/Member.js]: The purpose of this call is to just return the url (string). Just return a string - we usually try to hide the "request/response" logic from the Member.js layer anyway, especially if you only need to return a single string.

File: tpp/src/main/Member.js [hunk 5]
```diff
@@ -483,4 +486,42 @@ export default class Member extends CoreMember {
...
+    }
+
+    /**
+     * Forward the callback from the bank (after user authentication) to Token.
+     *
+     * @param bankId {string} Bank Id
+     * @param query {string} HTTP query string
+     * @returns {OnBankAuthCallbackResponse} token request ID
+     */
+    onBankAuthCallback(bankId: string, query: string): Promise<OnBankAuthCallbackResponse> {
```
Comment (gianluca-token) [inline tpp/src/main/Member.js]: Return a string

File: tpp/src/main/Member.js [hunk 6]
```diff
@@ -483,4 +486,42 @@ export default class Member extends CoreMember {
...
+        });
+    }
+
+    /**
+     * Get the raw consent from the bank associated with a token.
+     *
+     * @param tokenId {string} Token Id
+     * @returns {GetRawConsentResponse} raw consent
+     */
+    getRawConsent(tokenId: string): Promise<GetRawConsentResponse> {
```
Comment (gianluca-token) [inline tpp/src/main/Member.js]: Return a string

File: core/src/types.js [hunk 7]
```diff
@@ -340,3 +340,15 @@ export type VerifyEidasResponse = {
   status: KonsentusVerificationStatus,
   statusDetails: string,
 };
+
+export type GetBankAuthUrlResponse = {
```
Comment (gianluca-token) [inline core/src/types.js]: rm?

File: tpp/src/http/AuthHttpClient.js [hunk 8]
```diff
@@ -401,6 +401,50 @@ class AuthHttpClient extends CoreAuthHttpClient {
...
+     * Forward the callback from the bank (after user authentication) to Token.
+     *
+     * @param bankId {string} bank Id
+     * @param query {string} HTTP query string
+     * @returns {Object} response to the api call
+     */
+    async onBankAuthCallback(bankId, query){
+        const request = {
+            method: 'post',
+            url: `/banks/${bankId}/callback?${query}`,
```
Comment (gianluca-token) [inline tpp/src/http/AuthHttpClient.js]: should be `query=${query}`?

Comment (grekinsky) [reply tpp/src/http/AuthHttpClient.js]: yes I actually added that to the member method but you are right, it makes more sense here, I'm going to change it

**Pull request merged**